optimum/habana/transformers/trainer.py

```diff
@@ -867,6 +846,28 @@ def hpu_deepspeed_checkpointing(function, *checkpoint_args, use_reentrant: Optio
         self._globalstep_last_logged = self.state.global_step
         self._zero_model_grad(model)
         _grad_norm: Optional[float] = None
+        _should_compute_grad_norm: bool = not self.accelerator.distributed_type == GaudiDistributedType.DEEPSPEED and (
+            # Gradient clipping
+            args.max_grad_norm is not None and args.max_grad_norm > 0
+        )
+
+        # attn_softmax_bf16 and use_flash_attention are enabled only for llama, qwen2, starcoder2 and gemma
```

Comment:
Code is ok as is, however, would it be better to put this in a function which returns the setting for _should_update_inputs ? the list of models may expand in the future, so may be better to set it somewhere else?

Reply:
Thank you for your feedback. I agree that encapsulating the logic in a function could improve maintainability, especially as the list of models might expand in the future. I’ve refactored the code accordingly